File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    for a1,a2 in zip(arr1,arr2):
        st=''
        map1=(bin(a1)[2:].zfill(n))
        map2=(bin(a2)[2:].zfill(n))
        for i in range(n):
            if map1[i]=='1' or map2[i]=='1' :
                st+='#'
            else:
                st+=' '
        answer.append(st)
    return answer

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
import java.util.*;

class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        for(int i=0;i<n;i++){ //지도 생성
            String map1=Integer.toBinaryString(arr1[i]);
            map1=String.format("%"+n+"s",map1).replace(' ','0');
            String map2=Integer.toBinaryString(arr2[i]);
            map2=String.format("%"+n+"s",map2).replace(' ','0');
          
            
            String sumMap="";
            for(int j=0;j<n;j++){
                if(map1.charAt(j)=='0'&&map2.charAt(j)=='0') sumMap+=" ";
                else sumMap+="#";
            }
            answer[i]=sumMap;
        }
        
        
        
        return answer;
    }
}